### ISSUE TITLE
Wrong error reported - Remove this unnecessary 'using'

### ISSUE THREAD

**Issue by orantwolinkme** [SonarSource/sonar-dotnet#3390]
### Description

Sonar C# is asking to remove the using statement but it's required for System.Uri﻿

### Repro steps
```C#
using System;

namespace Users
{
    public class UserRecord
    {
        public UserRecord(string email, Uri photoUrl)
        {
            Email = email;
            PhotoUrl = photoUrl;
        }
        public string Email { get; }
        public Uri PhotoUrl { get; }
    }
}
```
### Expected behavior

No error reported

### Actual behavior

Remove this unnecessary 'using'.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* Sonar C# 8.7
* Rider 2020.1
* dotnet core 3.1
* Windows 10 2004

**Comment by costin-zaharia-sonarsource** [comment]
Hi @orantwolinkme 

I've tried to reproduce this issue with Sonar C# 8.7 but I didn't had any luck.

You can find below the sample I've used (there might be something that I'm missing): [Repro3390.zip](https://github.com/SonarSource/sonar-dotnet/files/4812103/Repro3390.zip)

Could you please create a small reproduction project for this and add it to this issue?

Thanks,
Costin

**Comment by andrei-epure-sonarsource** [comment]
Closing as we didn't get any reply.
